src/components/Calculator.js

```diff
@@ -1,78 +1,72 @@
-/* eslint-disable */
-import React from 'react';
+import React, { useState } from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import PropTypes from 'prop-types';
 import calculate from '../logic/calculate';
 import './Calculator.css';
 
-class Button extends React.Component {
- constructor(props) {
-    super(props);
-     this.processRecievedInput = this.processRecievedInput.bind(this);
- }
- processRecievedInput (e) {
-    this.props.recievedInput(e.target.textContent);
- }
-render () {
-    return (
-        <button className={this.props.class} onClick={this.processRecievedInput}>{this.props.name}</button>
-    );
-}
-}
+const Button = (props) => {
+  const { name, myClass, recievedInput } = props;
+  const processRecievedInput = (e) => {
+    recievedInput(e.target.textContent);
+  };
 
-class Paragraph extends React.Component {
-constructor(props) {
-    super(props);
-}
-
-render () {
-        let result = '';
-        if (this.props.total) {
-            result += this.props.total + ' ';
-        }
-        
-        if (this.props.operation) {
-            result += this.props.operation + ' ';
-        }
-        if (this.props.next) {
-            result += this.props.next + ' ';
-        }
-        
-    return (
-        <p>{result}</p>
-        );
-}
-}
+  return (
+    <button type="submit" className={myClass} onClick={processRecievedInput}>{name}</button>
+  );
+};
+Button.propTypes = {
+  name: PropTypes.string.isRequired,
+  myClass: PropTypes.string.isRequired,
+  recievedInput: PropTypes.string.isRequired,
+};
+const Paragraph = (props) => {
+  let result = '';
+  const { total, operation, next } = props;
+  if (total) {
+    result += `${total} `;
+  }
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-        total: null,
-        next: null,
-        operation: null};
-    this.passInput = this.passInput.bind(this);
+  if (operation) {
+    result += `${operation} `;
   }
-  passInput(buttonName) {
-    this.setState(calculate(this.state, buttonName));
+  if (next) {
+    result += `${next} `;
+  }
+  return (
+    <p>{result}</p>
+  );
+};
+Paragraph.propTypes = {
+  total: PropTypes.string.isRequired,
+  operation: PropTypes.string.isRequired,
+  next: PropTypes.string.isRequired,
+};
+const Calculator = () => {
+  const initialS = {
+    total: null,
+    next: null,
+    operation: null,
+  };
+  const [s, setS] = useState(initialS);
+  const passInput = (buttonName) => {
+    setS(calculate(s, buttonName));
```

Comment:
Kindly delete console.log(line 53) in your code to follow best practices

Reply:
@ahmetbozaci thanks a lot for mentioning that solved